pkg/api/user.go

```diff
@@ -284,7 +284,34 @@ func (hs *HTTPServer) handleUpdateUser(ctx context.Context, cmd user.UpdateUserC
 	return response.Success("User updated")
 }
 
-func (hs *HTTPServer) verifyEmailUpdate(ctx context.Context, email string, field user.UpdateEmailActionType, usr *user.User) response.Response {
+func (hs *HTTPServer) StartEmailVerificaton(c *contextmodel.ReqContext) response.Response {
+	namespace, id := c.SignedInUser.GetNamespacedID()
+	if !identity.IsNamespace(namespace, identity.NamespaceUser) {
+		return response.Error(http.StatusBadRequest, "Only users can verify their email", nil)
+	}
+
+	if c.SignedInUser.GetEmailVerified() {
+		// email is already verified so we don't need to trigger the flow.
+		return response.Respond(http.StatusNotModified, nil)
+	}
+
+	userID, err := strconv.ParseInt(id, 10, 64)
```

Comment:
```suggestion
	userID, err := identity.IntIdentifier(namespace, id)
```

Reply:
Yeah I know of that function and I choose not to use that one on purpose to avoid doing a duplicated check for namespace. I think we should rework that part. Right now I already do a namespace check on line 289. If a call identity.IntIdentifier(namespace, id) it would do that again which is a bit annoying. We also have identity.UserIdentifier but that one checks for service-accounts and users so that could not replace them both either..